src/Functions/csvToJsonString.cpp

```diff
@@ -0,0 +1,408 @@
+#include "Columns/ColumnNullable.h"
+#include "Columns/ColumnString.h"
+#include "FunctionHelpers.h"
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+extern const int BAD_ARGUMENTS;
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+namespace
+{
+/** Converts a CSV string to a JSON string.
+  * The function takes two or three arguments:
+  * - fieldNames: A comma-separated string of field names (must be a string constant)
+  * - csvValue: A comma-separated string of values (nullable)
+  * - optional key/value parameters to control the CSV parsing behavior
+  *
+  * The function returns a JSON string where the field names are mapped to their corresponding values.
+  * Unless disabled, the function attempts to detect the types of values and convert them appropriately:
+  * - Numbers are converted to JSON numbers
+  * - "true" and "false" are converted to JSON booleans
+  * - All other values are treated as strings
+  * 
+  * Example:
+  * csvToJSONString('name,age', 'John,42') => '{"name":"John","age":42}'
+  * csvToJSONString('name|age', 'John|42') SETTINGS format_csv_delimiter = '|' => '{"name":"John","age":42}'
+  * csvToJSONString('name$age', 'John$42', 'delimiter="$") => '{"name":"John","age":42}'
+  */
+class FunctionCsvToJsonString final : public IFunction
+{
+private:
+    ContextPtr context;
+
+    /** Splits a CSV string into a vector of strings according to the provided settings.
+      * @param s The input CSV string to split
+      * @param vectorBuffer A pre-allocated vector to store the split fields
+      * @param settings CSV format settings that control parsing behavior (delimiter, quotes, etc)
+      */
+    static void splitCSV(const StringRef & s, Strings & vectorBuffer, const FormatSettings::CSV & settings)
+    {
+        ReadBufferFromMemory buf(s.data, s.size);
+        vectorBuffer.clear();
+
+        while (!buf.eof())
+        {
+            String field;
+            readCSVString(field, buf, settings);
+            vectorBuffer.push_back(field);
+
+            // Skip delimiter if not at end
+            if (!buf.eof() && *buf.position() == settings.delimiter)
+                ++buf.position();
+        }
+    }
+
+    /** Merges a vector of field names and values into a JSON String.
+      * @param fieldNames A vector of field names
+      * @param fieldValues A vector of field values
+      * @param settings CSV format settings to get the null representation
+      * @param dest A reference to the destination stringstream to store the JSON result
+      * @param detectTypes A boolean flag to determine if types should be detected and converted
+      */
+    static void mergeAsJSON(
+        const Strings & fieldNames,
+        const Strings & fieldValues,
+        const FormatSettings::CSV & settings,
+        std::stringstream & dest,
+        bool detectTypes)
+    {
+        const auto n = std::min(fieldNames.size(), fieldValues.size());
```

Comment:
Can the 2 sizes be different?? I guess yes when the data is garbage in, when we provide more data than col?
Don't we want to trace it we have a problem here?
because the extra data will be lost silently